Spells

Davoleo opened the pull request:
## Spell and their FXs
- [x] Scatter
- [x] Thaw
- [x] Disarm
- [x] ~~Scythe~~
- [x] Soft touch
- [x] Fall
- [x] Blades
- [x] Drizzle

## Spell Docs:
- [x] Scatter
- [x] Thaw
- [x] Disarm
- [x] Soft touch
- [x] Fall
- [x] Blades
- [x] Drizzle

Closes #35

Davoleo committed:
Scatter Spell

Davoleo commented:
I still need to fix the boolean method for Scatter spell, if you know a better way to check if you can place seeds please tell me, I don't know many forge methods

noobanidus committed:
Merge branch 'master' into dav-corner

noobanidus committed:
Update SpellScatter to use sendToAllTracking instead.

Davoleo committed:
Merge branch 'master' into dav-corner

Davoleo committed:
I really am too tired for this :P

duely commented:
I'll have a look at the scatter spell for you when I remember. Also, I'm stealing Scythe off you as I have a specific implementation and use-case in mind and I'd like to get it in ASAP but I don't want to force you to working on it!

Davoleo committed:
Scatter Spell work

Davoleo committed:
Merge remote-tracking branch 'origin/master' into dav-corner

Davoleo committed:
Thaw Spell

Davoleo committed:
Merge branches 'dav-corner' and 'master' of https://github.com/EpicSquid/Roots into dav-corner

Davoleo committed:
Merge remote-tracking branch 'origin/master' into dav-corner

Davoleo committed:
Disarm Spell

Davoleo committed:
Small naming and localization fixes

Davoleo commented:
Should I still work on the "Shearing/Shed" spell @duely 
I've seen it's strikethrough in the issue

Davoleo committed:
Entities now drop their weapons

Davoleo committed:
Register Disarm FX Packet | Add chance to drop for each item

duely commented:
Skip "shed" in the sense of making chickens drop feathers; a shearing spell simply to aoe-shear sheep is not outside of reason but I think that would be up to @EpicSquid

Davoleo commented:
Oki

Il mer 10 apr 2019, 03:53 Jon <notifications@github.com> ha scritto:

> Skip "shed" in the sense of making chickens drop feathers; a shearing
> spell simply to aoe-shear sheep is not outside of reason but I think that
> would be up to @EpicSquid <https://github.com/EpicSquid>
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/EpicSquid/Roots/pull/97#issuecomment-481502769>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AXU0JKQgvneVs3j3htRdd_k7bUtmWZEMks5vfUQugaJpZM4cUjjl>
> .
>

Davoleo committed:
Merge branch 'master' into dav-corner

Davoleo committed:
Soft Touch Spell

Davoleo committed:
Soft Touch FX (obv broken)

Davoleo committed:
Merge branch 'master' into dav-corner

Davoleo committed:
Merge branch 'master' into dav-corner

Davoleo committed:
Update dav-corner

Davoleo committed:
Fix registry derps | Fix Scatter Spell FX

Davoleo committed:
Improve Scatter Spell | Fix Broken Disarm FX

duely commented:
I also made the runic shears do aoe shearing.

Davoleo committed:
Update dav-corner

Davoleo committed:
Fix missing imports

Davoleo committed:
Fix Soft Touch localization key and Visual FX

Davoleo committed:
Disarm Balance tweaks | Fall Spell

Davoleo committed:
Merge branch 'master' into dav-corner

Davoleo committed:
Some fixes regarding Thaw Spell

Davoleo committed:
Fall Spell Works, yay

- I have a really cool idea for its FX :thinking:

Davoleo committed:
Fall FX

Davoleo committed:
Hopefully fix my workspace -.-

+ Spell Fixes

Davoleo committed:
Update Dav-Corner

Davoleo committed:
Implement Properties into spells | Fix some merge mistakes

Davoleo committed:
Merge branch 'master' into dav-corner

Davoleo commented:
> Early-ish game Spell idea "Mark": You cast the spell on any surface and It'll place a small glowing flower petal. Optional feature?: can give spelldust a dye before equipping to staff and petal will change to that color(default green?).  (Used to mark paths when exploring caves/nether anywhere so you dont get lost.)

_Cit Kalaxi_

Davoleo committed:
Merge remote-tracking branch 'origin/master' into dav-corner

Davoleo committed:
Fix Merge Conflicts | Update spell to override init

Davoleo committed:
Fix Spell Block Placement | Basic Mark Spell Class

duely commented:
It definitely should be a no-cost spell in my opinion.

Davoleo committed:
More Mark Spell Work

Davoleo committed:
Model | Texture | Adjust Blockstate

Davoleo committed:
IBlockColor implementation | No Mark Spell Cost

Davoleo committed:
Update 'dav-corner'

Conflicts:
	src/main/java/epicsquid/roots/block/runes/BlockTrample.java
	src/main/java/epicsquid/roots/entity/spell/EntityBoost.java
	src/main/java/epicsquid/roots/recipe/RunicShearRecipe.java
	src/main/java/epicsquid/roots/recipe/RunicShearRecipes.java
	src/main/java/epicsquid/roots/recipe/conditions/ConditionItems.java
	src/main/java/epicsquid/roots/ritual/RitualBase.java
	src/main/java/epicsquid/roots/ritual/RitualWildGrowth.java
	src/main/java/epicsquid/roots/ritual/conditions/ConditionItems.java
	src/main/java/epicsquid/roots/spell/SpellSenseAnimals.java
	src/main/java/epicsquid/roots/spell/SpellSenseDanger.java

Davoleo committed:
Merge branch 'master' into dav-corner

# Conflicts:
#	build.gradle
#	src/main/java/epicsquid/roots/integration/crafttweaker/tweaks/RitualTweaker.java

Davoleo committed:
Adapt spells to the new finalisation system

Davoleo committed:
Fix some conditions - Remove the color state

Davoleo committed:
Improve Random BlockPos Methods

Davoleo committed:
Fall and Blades Spells Work

duely commented on src/main/java/epicsquid/roots/util/RitualUtil.java:
```diff
@@ -12,30 +13,45 @@
+
+  public static BlockPos getRandomPosRadialXYZ(World world, BlockPos centerPos, int xRadius, int yRadius, int zRadius, Block... whitelistedBlocks)
+  {
+    BlockPos.MutableBlockPos pos = new BlockPos.MutableBlockPos(centerPos.getX() -xRadius, centerPos.getY() -yRadius, centerPos.getZ() -zRadius);
```
Mutable is mostly when you're iterating over a pre-defined area and you want to save on memory usage by reusing the same BlockPos *object* instance but changing its X/Y/Z values. You also need to make sure that you use "toImmutable" or whatever it's called now before passing it back unless you're explicitly returning a Mutable Block Pos.

In this case, `pos.add` actually changes `pos` (I think...) so you don't even need to declare the second object. I don't know that it really saves much though, because all you're doing is reducing the instantiation of a second BlockPos object.

`BlockPos.getAllInBoxMutable` is a really good example of its usage. It creates one object and just increments/decrements the x, y and z values as necessary instead of creating an object for each time `.next()` is called. If you're doing a large space this can save on object creation costs.

</random waffle>

Davoleo replied:
oki, I'll check it out later

Davoleo committed:
Add some more transmutations to Iced Touch

Davoleo committed:
Thaw GFX

Davoleo committed:
Revert to non-mutable BlockPos

Davoleo committed:
First Drizzle Spell Classes

+ gave up on Mark spell

Davoleo committed:
Merge remote-tracking branch 'origin/master' into dav-corner

Davoleo committed:
Merge branch 'master' into dav-corner

Davoleo committed:
Drizzle Spell Logic

Davoleo committed:
Drizzle Spell GFX  (Half way done)

- I need to find some way to slow it down

Davoleo committed:
Update branch

Davoleo committed:
Fix Drizzle Effect

Davoleo committed:
Scatter Spell Documentation

Davoleo committed:
Thaw Spell Docs

Davoleo committed:
Disarm Docs

+ Tweaked the spell cost

Davoleo committed:
Soft Touch Spell Docs

Davoleo committed:
Fall Spell Docs

+ Tweaked Spell cost

Davoleo committed:
Blades Spell Docs

Davoleo committed:
Drizzle Spell Docs

Davoleo committed:
Call the FX Packet for all the blocks within the radius

Davoleo committed:
Drizzle Spell Visual FX

Davoleo commented:
This is finally ready, asking @duely for a review
